### ISSUE TITLE
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: socket hang up

### ISSUE THREAD

**Issue by wangzhenmiao** [appium/appium#12931]
## The problem

my appium case has error occasionally
error is:socket hang up
if i reboot appnium-server and my device ,it may disappear
but i want my case stable,i need you help,Thanks in advance！

## Environment

* Appium version (or git revision) that exhibits the issue:
1.14.0
* Node.js version (unless using Appium.app|exe):
v11.14.0
* Npm or Yarn package manager:
6.9.0
* Mobile platform/version under test:
android 8.1.0

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs

[debug] [W3C (b8248a78)] Calling AppiumDriver.findElement() with args: ["id","myid","b8248a78-f2c1-4e2f-957b-7e3276656458"]
[debug] [BaseDriver] Valid locator strategies for this request: xpath, id, class name, accessibility id, -android uiautomator
[debug] [BaseDriver] Waiting up to 0 ms for condition
[debug] [WD Proxy] Matched '/element' to command name 'findElement'
[debug] [WD Proxy] Proxying [POST /element] to [POST http://localhost:8200/wd/hub/session/8841ba95-45cf-4ab3-b02c-bfdd678cd4b9/element] with body: {"strategy":"id","selector":"myid","context":"","multiple":false}
[WD Proxy] Got an unexpected response: {"code":"ECONNRESET"}
[debug] [W3C (b8248a78)] Encountered internal error running command: UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: socket hang up
[debug] [W3C (b8248a78)]     at JWProxy.command (/usr/local/lib/node_modules/appium/node_modules/_appium-base-driver@3.17.1@appium-base-driver/lib/jsonwp-proxy/proxy.js:248:13)
[HTTP] <-- POST /wd/hub/session/b8248a78-f2c1-4e2f-957b-7e3276656458/element 500 14 ms - 557

**Comment by mykola-mokhnach** [comment]
Make sure the usbmuxd package is up to date. There was a known bug there, which was causing intermittent disconnects. Also, it makes sense to verify the device logs

**Comment by wangzhenmiao** [comment]
> Make sure the usbmuxd package is up to date. There was a known bug there, which was causing intermittent disconnects. Also, it makes sense to verify the device logs

my device is android.i first heard usbmuxd, is it used in ios?

**Comment by mykola-mokhnach** [comment]
Oh, I thought it is iOS. On Android it makes sense to check if the SDK is up to date. 
The logcat output can also sometimes contain hints on why disconnects are happening

Possible duplicate of https://github.com/appium/appium/issues/12909

**Comment by wangzhenmiao** [comment]
> Oh, I thought it is iOS. On Android it makes sense to check if the SDK is up to date.
> The logcat output can also sometimes contain hints on why disconnects are happening
> 
> Possible duplicate of #12909

thanks a lot.I continue to work hard.Haha~

**Comment by rogersan88** [comment]
hola , me esta pasando el mismo problema , has podido solucionarlo?

**Comment by mykola-mokhnach** [comment]
Without the detailed logcat output there is nothing more to do on this issue

**Comment by wangzhenmiao** [comment]
my partner tell me : when he change to anther phone ,the problem resolved.
vivo is always with this problem, we can use huawei, xiaomi and so on, just try!

**Comment by wirahitaputramas** [comment]
Hello guys, I have faced the same problem, but now I just find the reason & also the solution. I realized that the error occurred when I made a test case that contains both "close server session" & "open new server session" command. I Think when running that kind of test case, the two different appium sessions overlap with each other, and then the error occurred. So my solution to this problem was very simple because I just separate "close server session" & "open new server session" command into two different test case, and then boom, the error has not occurred until now
